src/main/java/io/stargate/sgv2/jsonapi/service/embedding/DataVectorizer.java

```diff
@@ -125,13 +130,23 @@ public void vectorizeUpdateClause(UpdateClause updateClause) {
     updateVectorize(setOnInsertNode);
     final ObjectNode unsetNode = updateClause.updateOperationDefs().get(UpdateOperator.UNSET);
     if (unsetNode != null && unsetNode.has(DocumentConstants.Fields.VECTOR_EMBEDDING_TEXT_FIELD)) {
+      if (unsetNode.has(DocumentConstants.Fields.VECTOR_EMBEDDING_FIELD)) {
+        throw new JsonApiException(
```

Comment:
I think you can simplify this to

    throw new JsonApiException(ErrorCode.INVALID_USAGE_OF_VECTORIZE);

since `ErrorCode.getMessage()` is used without anything else as message.